services/replicator/replicator.go

```diff
@@ -568,31 +568,280 @@ func (r *Replicator) deleteDestinationRemoteCall(zone string, logger bark.Logger
+		return err
+	}
+
+	// for all the zones of current tenancy
+	for _, zone := range r.allZones[r.tenancy] {
+		// skip local zone
+		if strings.EqualFold(zone, r.localZone) {
+			continue
+		}
+
+		// call remote replicators in a goroutine. Errors can be ignored since reconciliation will fix the inconsistency eventually
+		go r.createConsumerGroupRemoteCall(zone, lclLg, createRequest)
+	}
+
+	return nil
+}
+
+func (r *Replicator) createConsumerGroupRemoteCall(zone string, logger bark.Logger, createRequest *shared.CreateConsumerGroupUUIDRequest) error {
+	// acquire remote zone replicator thrift client
+	client, err := r.replicatorclientFactory.GetReplicatorClient(zone)
+	if err != nil {
+		r.m3Client.IncCounter(metrics.ReplicatorCreateRmtCgUUIDScope, metrics.ReplicatorFailures)
+		logger.WithFields(bark.Fields{
+			common.TagErr:      err,
+			common.TagZoneName: common.FmtZoneName(zone),
+		}).Error(`Get remote replicator client failed`)
+		return err
+	}
+
+	// send to remote zone replicator
+	ctx, cancel := thrift.NewContext(remoteReplicatorCallTimeOut)
+	defer cancel()
+	_, err = client.CreateConsumerGroupUUID(ctx, createRequest)
+	if err != nil {
+		r.m3Client.IncCounter(metrics.ReplicatorCreateRmtCgUUIDScope, metrics.ReplicatorFailures)
+		logger.WithFields(bark.Fields{
+			common.TagErr:      err,
+			common.TagZoneName: common.FmtZoneName(zone),
+		}).Error(`Create remote cg call failed`)
+		return err
+	}
+
 	return nil
 }
 
 // UpdateConsumerGroup updates consumer group at local zone, expect to be called by remote replicator
 func (r *Replicator) UpdateConsumerGroup(ctx thrift.Context, updateRequest *shared.UpdateConsumerGroupRequest) (*shared.ConsumerGroupDescription, error) {
-	return nil, nil
+	r.m3Client.IncCounter(metrics.ReplicatorUpdateCgScope, metrics.ReplicatorRequests)
+
+	cgDesc, err := r.metaClient.UpdateConsumerGroup(ctx, updateRequest)
+	if err != nil {
+		r.logger.WithFields(bark.Fields{
+			common.TagCnsPth: common.FmtCnsPth(updateRequest.GetConsumerGroupName()),
+			common.TagDstPth: common.FmtDstPth(updateRequest.GetDestinationPath()),
+			common.TagDst:    common.FmtDst(cgDesc.GetDestinationUUID()),
+			common.TagErr:    err,
+		}).Error(`Error updating cg`)
+		r.m3Client.IncCounter(metrics.ReplicatorUpdateCgScope, metrics.ReplicatorFailures)
+		return nil, err
+	}
+
+	r.logger.WithFields(bark.Fields{
+		common.TagCnsPth: common.FmtCnsPth(updateRequest.GetConsumerGroupName()),
+		common.TagDstPth: common.FmtDstPth(updateRequest.GetDestinationPath()),
+		common.TagDst:    common.FmtDst(cgDesc.GetDestinationUUID()),
+		common.TagDLQID:  common.FmtDLQID(cgDesc.GetDeadLetterQueueDestinationUUID()),
+		`IsMultiZone`:    cgDesc.GetIsMultiZone(),
+		`ActiveZone`:     cgDesc.GetActiveZone(),
+	}).Info(`Updated cg`)
+	return cgDesc, nil
 }
 
 // UpdateRemoteConsumerGroup propagate update to multiple remote zones, expect to be called by local zone services
```

Comment:
the plan is to limit destination/cg CRUD operation in authoritative zone only, although we haven't implemented that yet. I think 'local zone' here implies authoritative zone, I've added some comment to clarify